source/agora/network/NetworkManager.d

```diff
@@ -368,6 +369,9 @@ public class NetworkManager
     /// Clock instance
     protected Clock clock;
 
+    /// Stats server
+    protected StatsServer stats_server;
+
```

Comment:
This whole diff feels like it belongs more to the `FullNode`, doesn't it ?

Reply:
when I was implementing the Name Registry I was told not to put network related code into FullNode, that is why we put    
private NameRegistryAPI registry_client;
into NetworkManager

should I move stats_server into the FullNode?